packages/controllers/src/snaps/endowments/cronjob.test.ts

```diff
@@ -0,0 +1,223 @@
+      expect(
+        getCronjobCaveatMapper({
+          jobs: [
+            {
+              expression: '* * * * *',
+              request: {
+                method: 'exampleMethodOne',
+                params: ['p1'],
+              },
+            },
+            {
+              expression: {
+                minute: '*',
+                hour: '*',
+                dayOfMonth: '*',
+                month: '*',
+                dayOfWeek: '*',
+              },
+              request: {
+                method: 'exampleMethodTwo',
+                params: ['p1', 'p2', 'p3'],
+              },
+            },
+          ],
+        }),
+      ).toStrictEqual({
+        caveats: [
+          {
+            type: SnapCaveatType.SnapCronjob,
+            value: {
+              jobs: [
+                {
+                  expression: '* * * * *',
+                  request: {
+                    method: 'exampleMethodOne',
+                    params: ['p1'],
+                  },
+                },
+                {
+                  expression: {
+                    minute: '*',
+                    hour: '*',
+                    dayOfMonth: '*',
+                    month: '*',
+                    dayOfWeek: '*',
+                  },
+                  request: {
+                    method: 'exampleMethodTwo',
+                    params: ['p1', 'p2', 'p3'],
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      });
+    });
+  });
+});
+
+describe('validateCronjobCaveat', () => {
+  it('should not throw an error when provided specification is valid', () => {
+    const caveat: Caveat<string, any> = {
+      type: SnapCaveatType.SnapCronjob,
+      value: {
+        jobs: [
+          {
+            expression: {
+              minute: '*',
+              hour: '*',
+              dayOfMonth: '*',
+              month: '*',
+              dayOfWeek: '*',
+            },
+            request: {
+              method: 'exampleMethodOne',
+              params: ['p1'],
+            },
+          },
+          {
+            expression: '* * * * *',
+            request: {
+              method: 'snapMethod',
+              params: [],
+            },
+          },
+        ],
+      },
+    };
+
+    expect(() => validateCronjobCaveat(caveat)).not.toThrow();
+  });
+
+  it('should throw if caveat has no proper value', () => {
+    const caveat: Caveat<string, any> = {
+      type: SnapCaveatType.SnapCronjob,
+      value: {},
+    };
+
+    expect(() => validateCronjobCaveat(caveat)).toThrow(
+      `Expected a plain object.`,
+    );
+  });
+
+  it('should throw an error when cron specification is missing', () => {
+    const caveat: Caveat<string, any> = {
+      type: SnapCaveatType.SnapCronjob,
+      value: {
+        jobs: [
+          {
+            expression: '* * * * *',
+            request: undefined as any,
```

Comment:
We should prefer `@ts-expect-error` over casting to `any` for testing, in my opinion.

```suggestion
            // @ts-expect-error Invalid type.
            request: undefined,
```

Reply:
Changed. It actually let me use just `undefined` in this case.